pymatviz/io.py

```diff
@@ -109,13 +110,13 @@ def save_fig(
         if path.lower().endswith(".svelte"):
             # insert {...$$props} into top-level div to be able to post-process and
             # style plotly figures from within Svelte files
-            with open(path) as file:
+            with open(path, encoding="utf-8") as file:
```

Comment:
I think `encoding="utf-8"` can be [considered the default](https://docs.python.org/3/library/locale.html#locale.getencoding). I would only specify if we encounter a case where it matters.

Reply:
Sorry I don't mean to include you into the conversation so early (thus it's a draft). But it's great discussing with you as always. I'm still busying clean up my Paper for ArXiv, and am just work on this from time to time...

Not exactly I would say, from the [page](https://docs.python.org/3/library/locale.html#locale.getencoding) you mentioned there is:
> On Android and VxWorks, return "utf-8".
> On Unix, return the encoding of the current [LC_CTYPE](https://docs.python.org/3/library/locale.html#locale.LC_CTYPE) locale. Return "utf-8" if nl_langinfo(CODESET) returns an empty string: for example, if the current LC_CTYPE locale is not supported.
> On Windows, return the ANSI code page.

Given the popularity of Windows OS, it might be better to explicitly declare the encoding for robustness (I have seen quite some weird issues related to encoding as I'm working on Windows and Linux at the same time), what do you think?

Also [`pylint`](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unspecified-encoding.html) encourages explicit encoding by giving this warning if not:

> unspecified-encoding / W1514
> Message emitted:
> Using open without explicitly specifying an encoding
> 
> Description:
> It is better to specify an encoding when opening documents. Using the system default implicitly can create problems on other operating systems. See https://peps.python.org/pep-0597/